### PR TITLE
Add UI package + docs

### PR DESCRIPTION
**Pull request by narph**
- add comments
- add contribution files and more docs
- add UI package

### CODE REVIEW

File: src/packages/Elastic.ElasticAgent-windows-arm/Strings/Resources.resjson [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "displayName": "Elastic Agent for Windows",
+  "publisherDisplayName": "Elastic N.V.",
+  "summary": "Elastic Agent for Windows is a lightweight data shipper that collects system logs and metrics from your virtual machines and stores them in Elasticsearch.",
+  "description": "STo configure the agent you’ll need to set your Elastic Cloud ID along with your Elasticsearch Service user credentials. Don’t have an account? Sign up to Elastic Cloud if you are a new user. Elastic Agent lets you get started with monitoring your infrastructure health in seconds: Collect logs, metrics and APM data in once place. Out of the box integrations for more than 50 applications, such as MSSQL, Zookeeper, Memchached, and more. Powerful Kibana UI apps for infrastructure and application observability. Flexible alerting and integration with 3pp ticketing systems, e.g. PagerDuty, Slack, Jira.",
+  "link1": "Shttps://www.elastic.co/guide/en/fleet/current/index.html"
```
Comment (v1v) [inline src/packages/Elastic.ElasticAgent-windows-arm/Strings/Resources.resjson:6]:
```suggestion
  "link1": "https://www.elastic.co/guide/en/fleet/current/index.html"
```

File: src/packages/Elastic.ElasticAgent-windows-arm/Strings/Resources.resjson [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "displayName": "Elastic Agent for Windows",
+  "publisherDisplayName": "Elastic N.V.",
+  "summary": "Elastic Agent for Windows is a lightweight data shipper that collects system logs and metrics from your virtual machines and stores them in Elasticsearch.",
+  "description": "STo configure the agent you’ll need to set your Elastic Cloud ID along with your Elasticsearch Service user credentials. Don’t have an account? Sign up to Elastic Cloud if you are a new user. Elastic Agent lets you get started with monitoring your infrastructure health in seconds: Collect logs, metrics and APM data in once place. Out of the box integrations for more than 50 applications, such as MSSQL, Zookeeper, Memchached, and more. Powerful Kibana UI apps for infrastructure and application observability. Flexible alerting and integration with 3pp ticketing systems, e.g. PagerDuty, Slack, Jira.",
```
Comment (v1v) [inline src/packages/Elastic.ElasticAgent-windows-arm/Strings/Resources.resjson:5]:
```suggestion
  "description": "To configure the agent you’ll need to set your Elastic Cloud ID along with your Elasticsearch Service user credentials. Don’t have an account? Sign up to Elastic Cloud if you are a new user. Elastic Agent lets you get started with monitoring your infrastructure health in seconds: Collect logs, metrics and APM data in once place. Out of the box integrations for more than 50 applications, such as MSSQL, Zookeeper, Memchached, and more. Powerful Kibana UI apps for infrastructure and application observability. Flexible alerting and integration with 3pp ticketing systems, e.g. PagerDuty, Slack, Jira.",
```

File: src/packages/Elastic.ElasticAgent.linux-arm/Strings/Resources.resjson [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "displayName": "Elastic Agent for Linux",
+  "publisherDisplayName": "Elastic N.V.",
+  "summary": "Elastic Agent for Linux is a lightweight data shipper that collects system logs and metrics from your virtual machines and stores them in Elasticsearch.",
+  "description": "STo configure the agent you’ll need to set your Elastic Cloud ID along with your Elasticsearch Service user credentials. Don’t have an account? Sign up to Elastic Cloud if you are a new user. Elastic Agent lets you get started with monitoring your infrastructure health in seconds: Collect logs, metrics and APM data in once place. Out of the box integrations for more than 50 applications, such as MSSQL, Zookeeper, Memchached, and more. Powerful Kibana UI apps for infrastructure and application observability. Flexible alerting and integration with 3pp ticketing systems, e.g. PagerDuty, Slack, Jira.",
+  "link1": "Shttps://www.elastic.co/guide/en/fleet/current/index.html"
```
Comment (v1v) [inline src/packages/Elastic.ElasticAgent.linux-arm/Strings/Resources.resjson:6]:
```suggestion
  "link1": "https://www.elastic.co/guide/en/fleet/current/index.html"
```

File: src/packages/Elastic.ElasticAgent.linux-arm/Strings/Resources.resjson [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "displayName": "Elastic Agent for Linux",
+  "publisherDisplayName": "Elastic N.V.",
+  "summary": "Elastic Agent for Linux is a lightweight data shipper that collects system logs and metrics from your virtual machines and stores them in Elasticsearch.",
+  "description": "STo configure the agent you’ll need to set your Elastic Cloud ID along with your Elasticsearch Service user credentials. Don’t have an account? Sign up to Elastic Cloud if you are a new user. Elastic Agent lets you get started with monitoring your infrastructure health in seconds: Collect logs, metrics and APM data in once place. Out of the box integrations for more than 50 applications, such as MSSQL, Zookeeper, Memchached, and more. Powerful Kibana UI apps for infrastructure and application observability. Flexible alerting and integration with 3pp ticketing systems, e.g. PagerDuty, Slack, Jira.",
```
Comment (v1v) [inline src/packages/Elastic.ElasticAgent.linux-arm/Strings/Resources.resjson:5]:
```suggestion
  "description": "To configure the agent you’ll need to set your Elastic Cloud ID along with your Elasticsearch Service user credentials. Don’t have an account? Sign up to Elastic Cloud if you are a new user. Elastic Agent lets you get started with monitoring your infrastructure health in seconds: Collect logs, metrics and APM data in once place. Out of the box integrations for more than 50 applications, such as MSSQL, Zookeeper, Memchached, and more. Powerful Kibana UI apps for infrastructure and application observability. Flexible alerting and integration with 3pp ticketing systems, e.g. PagerDuty, Slack, Jira.",
```

File: README.md [hunk 5]
```diff
@@ -1,7 +1,10 @@
-# Elastic Agent VM extension
+# Elastic Agent VM extensions
 
-The ElasticAgent VM extension is a small application that provides post-deployment configuration and automation on Azure VMs.
-Once installed, it will download the elastic agent artifacts, install the elastic agent on the virtual machine, enroll it to Fleet and then start the agent service.
+ElasticAgent.windows for Windows systems
+ElasticAgent.linux for Linux systems
+
+The ElasticAgent VM extensions are small applications that provides post-deployment configuration and automation on Azure VMs.
```
Comment (v1v) [inline README.md:6]:
```suggestion
The Elastic Agent VM extensions are small applications that provide post-deployment configuration and automation on Azure VMs.
```

File: README.md [hunk 6]
```diff
@@ -1,7 +1,10 @@
...
+Once installed, the ElasticAgent VM extension will download the elastic agent artifacts, install the elastic agent on the virtual machine, enroll it to Fleet and then start the agent service.
```
Comment (v1v) [inline README.md:7]:
```suggestion
Once installed, the ElasticAgent VM extension will download the Elastic Agent artifacts, install the elastic agent on the virtual machine, enroll it to Fleet and then start the agent service.
```

File: build/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+## Handler
+
+The handler implements the Windows Azure Agent defined protocol to integrate with the Azure ecosystem. 
+
+Handler requirements:
+
+1. Handler Packaging – The Handler should be packaged as a zip file. This zip package should contain all the binaries related to the handler and HandlerManifest. This package needs to be registered with the Azure image repository. Azure image repository is responsible for managing all versions of all the handlers that are registered with the Azure ecosystem.
+
+2. Handler Environment - Handler needs the capability to read the environment file in the format that Azure Agent defines. The environment file defines the locations of various files and folders that the handler needs to use for reading configuration and writing back heartbeat and status.
```
Comment (v1v) [inline build/CONTRIBUTING.md:15]:
```suggestion
2. Handler Environment - The handler needs the capability to read the environment file in the format that Azure Agent defines. The environment file defines the locations of various files and folders that the handler needs to use for reading configuration and writing back heartbeat and status.
```

File: build/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+- handler - contains the windows/linux related scripts that need to install needed dependencies on the vm, download the elastic agent install it and enroll it to Fleet
+- packages - contains the UI and miscellaneous installation scripts 
+
+## Handler
+
+The handler implements the Windows Azure Agent defined protocol to integrate with the Azure ecosystem. 
+
+Handler requirements:
+
+1. Handler Packaging – The Handler should be packaged as a zip file. This zip package should contain all the binaries related to the handler and HandlerManifest. This package needs to be registered with the Azure image repository. Azure image repository is responsible for managing all versions of all the handlers that are registered with the Azure ecosystem.
```
Comment (v1v) [inline build/CONTRIBUTING.md:13]:
```suggestion
1. Handler Packaging – The handler should be packaged as a zip file. This zip package should contain all the binaries related to the handler and HandlerManifest. This package needs to be registered with the Azure image repository. Azure image repository is responsible for managing all versions of all the handlers that are registered with the Azure ecosystem.
```

File: build/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+Elastic Agent Handler Package structure
+
+Binaries will be composed of .cmd, .ps1 and .sh files.
+
+Install command:
+- should download and install the necessary dependencies for the elastic agent to be installed.
+
+Enable command:
+- will parse the Ellasticsearch and Kibana urls fromt he cloud ID
```
Comment (v1v) [inline build/CONTRIBUTING.md:51]:
```suggestion
- will parse the Elasticsearch and Kibana urls from the cloud ID
```

File: build/CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+- will ping the ES cluster to retrieve the stack version
+- will download the elastic agent with the selected stack version
+- will install the elastic agent, enroll it to Fleet and start it
```
Comment (v1v) [inline build/CONTRIBUTING.md:54]:
```suggestion
- will ping the ES cluster to retrieve the stack version
- will download the Elastic Agent with the selected stack version
- will install the Elastic Agent, enroll it to Fleet and start it
```

File: build/CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+Will also handled cases like update configuration or start vm
+
+Disable command:
+ - Should only stop the service (cmd documented in elastic agent docs)
+
+Uninstall command:
+- Should uninstall the service. For deb and rpm, additional effort is made in order to unenroll the agent (api call), stop the agent and remove the elastic agent folders
```
Comment (v1v) [inline build/CONTRIBUTING.md:62]:
```suggestion
- Should uninstall the service. For deb and rpm, additional effort is made in order to unenroll the agent (api call), stop the agent and remove the Elastic Agent folders
```

File: build/CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+Update command:
+- Should re-install a new instance of elastic agent
```
Comment (v1v) [inline build/CONTRIBUTING.md:65]:
```suggestion
- Should re-install a new instance of Elastic Agent
```

File: build/CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+    }]
+---
+`logFolder` - contains the location where the handler should put its log files that might be needed to debug any customer issues. The advantage of putting log files under the folder directed by this location is that these files can be automatically retrieved from the customers VM by using a tool, without actually logging into the VM and copying them over manually. 
+`configFolder` - contains the location where the handler will get its configuration settings file. 
+`statusFolder` - contains the location where the handler is supposed to write back a file with a structured status of the current state of the work being done by the handler. heartbeatFile - this is the file that is used to communicate the heartbeat of the handler back to the Azure Agent. 
+`deploymentid` - this is the deployment id which the vm belongs to. This may change when a VM is captured/backed-up and restored. 
+`rolename` -  this is the role name of the vm. This may change when a VM is captured/backed up and restored. 
+`instance` - this is the instance name of the vm (can be same as role name)
+
+###Handler Configuration
```
Comment (v1v) [inline build/CONTRIBUTING.md:93]:
```suggestion
### Handler Configuration
```

File: src/packages/Elastic.ElasticAgent.linux-arm/Strings/Resources.resjson [hunk 14]
```diff
@@ -0,0 +1,7 @@
+{
+  "displayName": "Elastic Agent for Linux",
+  "publisherDisplayName": "Elastic N.V.",
+  "summary": "Elastic Agent for Linux is a lightweight data shipper that collects system logs and metrics from your virtual machines and stores them in Elasticsearch.",
+  "description": "To configure the agent you’ll need to set your Elastic Cloud ID along with your Elasticsearch Service user credentials. Don’t have an account? Sign up to Elastic Cloud if you are a new user. Elastic Agent lets you get started with monitoring your infrastructure health in seconds: Collect logs, metrics and APM data in once place. Out of the box integrations for more than 50 applications, such as MSSQL, Zookeeper, Memchached, and more. Powerful Kibana UI apps for infrastructure and application observability. Flexible alerting and integration with 3pp ticketing systems, e.g. PagerDuty, Slack, Jira.",
```
Comment (v1v) [inline src/packages/Elastic.ElasticAgent.linux-arm/Strings/Resources.resjson:5]:
```suggestion
  "description": "To configure the agent you’ll need to set your Elastic Cloud ID along with your Elasticsearch Service user credentials. Don’t have an account? Sign up to Elastic Cloud if you are a new user. Elastic Agent lets you get started with monitoring your infrastructure health in seconds: collect logs, metrics and APM data in once place. Out of the box integrations for more than 50 applications, such as MSSQL, Zookeeper, Memchached, and more. Powerful Kibana UI apps for infrastructure and application observability. Flexible alerting and integration with 3pp ticketing systems, e.g. PagerDuty, Slack, Jira.",
```

File: src/packages/Elastic.ElasticAgent-windows-arm/Strings/Resources.resjson [hunk 15]
```diff
@@ -0,0 +1,7 @@
+{
+  "displayName": "Elastic Agent for Windows",
+  "publisherDisplayName": "Elastic N.V.",
+  "summary": "Elastic Agent for Windows is a lightweight data shipper that collects system logs and metrics from your virtual machines and stores them in Elasticsearch.",
+  "description": "To configure the agent you’ll need to set your Elastic Cloud ID along with your Elasticsearch Service user credentials. Don’t have an account? Sign up to Elastic Cloud if you are a new user. Elastic Agent lets you get started with monitoring your infrastructure health in seconds: Collect logs, metrics and APM data in once place. Out of the box integrations for more than 50 applications, such as MSSQL, Zookeeper, Memchached, and more. Powerful Kibana UI apps for infrastructure and application observability. Flexible alerting and integration with 3pp ticketing systems, e.g. PagerDuty, Slack, Jira.",
```
Comment (v1v) [inline src/packages/Elastic.ElasticAgent-windows-arm/Strings/Resources.resjson:5]:
```suggestion
  "description": "To configure the agent you’ll need to set your Elastic Cloud ID along with your Elasticsearch Service user credentials. Don’t have an account? Sign up to Elastic Cloud if you are a new user. Elastic Agent lets you get started with monitoring your infrastructure health in seconds: collect logs, metrics and APM data in once place. Out of the box integrations for more than 50 applications, such as MSSQL, Zookeeper, Memchached, and more. Powerful Kibana UI apps for infrastructure and application observability. Flexible alerting and integration with 3pp ticketing systems, e.g. PagerDuty, Slack, Jira.",
```

File: README.md (hunk 5 above)
Comment (narph) [reply README.md]: I've handled all texts fixes

**Review by v1v — APPROVED** [review]

File: README.md [hunk 16]
```diff
@@ -1,7 +1,10 @@
-# Elastic Agent VM extension
+# Elastic Agent VM extensions
 
-The ElasticAgent VM extension is a small application that provides post-deployment configuration and automation on Azure VMs.
-Once installed, it will download the elastic agent artifacts, install the elastic agent on the virtual machine, enroll it to Fleet and then start the agent service.
+ElasticAgent.windows for Windows systems
+ElasticAgent.linux for Linux systems
+
+The ElasticAgent VM extensions are small applications that provide post-deployment configuration and automation on Azure VMs.
```
Comment (v1v) [inline README.md:6]:
```suggestion
The Elastic Agent VM extensions are small applications that provide post-deployment configuration and automation on Azure VMs.
```

File: README.md [hunk 17]
```diff
@@ -1,7 +1,10 @@
...
+Once installed, the ElasticAgent VM extension will download the Elastic Agent artifacts, install the Elastic Agent on the virtual machine, enroll it to Fleet and then start the agent service.
```
Comment (v1v) [inline README.md:7]:
```suggestion
Once installed, the Elastic Agent VM extension will download the Elastic Agent artifacts, install the Elastic Agent on the virtual machine, enroll it to Fleet and then start the agent service.
```

Comment (narph) [reply README.md:7]: the name of the vm extension is ElasticAgent but both options work

**Pull request merged**